Probably needs a more current version of Docker Engine to run the newer alpine / debian images (otherwise seccomp may be blocking some syscalls);

```
Step 22/51 : RUN make bin/compose-controller bin/compose-controller.test e2e-binary bin/installer bin/api-server bin/api-server.test bin/e2e_benchmark
 ---> Running in bb8946581e31
make: /bin/sh: Operation not permitted
make: /bin/sh: Operation not permitted
make: /bin/sh: Operation not permitted
make: /bin/sh: Operation not permitted
common.mk:21: unable to set BUILDTIME. Set the value manually
make: /bin/sh: Operation not permitted
make: /bin/sh: Operation not permitted
make: *** [Makefile:41: FORCE] Error 127
The command '/bin/sh -c make bin/compose-controller bin/compose-controller.test e2e-binary bin/installer bin/api-server bin/api-server.test bin/e2e_benchmark' returned a non-zero code: 2
make: *** [docker.Makefile:73: dockereng/kube-compose-controller] Error 2
```

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -17,7 +17,7 @@ aliases:
     - image: dockereng/kube-compose-ci-golang
       auth: *docker-login
   public-golang: &public-golang
-    - image: circleci/golang:1.13.0
+    - image: circleci/golang:1.16
       user: root
 
   e2e-kind-steps: &e2e-kind-steps

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -24,7 +24,7 @@ aliases:
     - attach_workspace:
         at: /root
     - setup_remote_docker:
-        version: 18.09.3
+        version: 20.10.11
     - run:
         name: "Prepare Kind and run e2e tests"
         command: |
@@ -49,7 +49,7 @@ aliases:
     - attach_workspace:
         at: /root
     - setup_remote_docker:
-        version: 18.09.3
+        version: 20.10.11
     - run:
         name: "Prepare Kind and run e2e benchmark"
         command: |
@@ -82,7 +82,7 @@ jobs:
     steps:
       - checkout
       - setup_remote_docker:
-          version: 18.09.3
+          version: 20.10.11
       - run:
           name: Login to docker hub
           command: echo "$DOCKER_PASS" | docker login -u "$DOCKER_USER" --password-stdin
@@ -112,7 +112,7 @@ jobs:
       - attach_workspace:
           at: /root
       - setup_remote_docker:
-          version: 18.09.3
+          version: 20.10.11
       - run: make -f docker.Makefile validate
   validate-vendor:
     docker: *public-golang
@@ -146,7 +146,7 @@ jobs:
       - attach_workspace:
           at: /root
       - setup_remote_docker:
-          version: 18.09.3
+          version: 20.10.11
       - run:
           name: Build images
           command: |
@@ -234,7 +234,7 @@ jobs:
       - attach_workspace:
           at: /root
       - setup_remote_docker:
-          version: 18.09.3
+          version: 20.10.11
       - run:
           name: Run unit tests
           command: make -f docker.Makefile test-unit
@@ -261,7 +261,7 @@ jobs:
       - attach_workspace:
           at: /root
       - setup_remote_docker:
-          version: 18.09.3
+          version: 20.10.11
       - deploy:
           name: push-to-hub
           command: |
@@ -292,7 +292,7 @@ jobs:
       - attach_workspace:
           at: /root
       - setup_remote_docker:
-          version: 18.09.3
+          version: 20.10.11
       - deploy:
           name: push-to-hub
           command: |

--- a/Dockerfile
+++ b/Dockerfile
@@ -12,6 +12,7 @@ WORKDIR /home/compose
 
 
 FROM ${BUILD_BASE} AS build
+ENV GO111MODULE=off
 
 RUN apk add --no-cache \
   coreutils \

--- a/Dockerfile
+++ b/Dockerfile
@@ -32,7 +32,7 @@ ARG KUBERNETES_VERSION
 ENV GITCOMMIT=$GITCOMMIT VERSION=$VERSION BUILDTIME=$BUILDTIME IMAGE_REPO_PREFIX=$IMAGE_REPO_PREFIX KUBERNETES_VERSION=$KUBERNETES_VERSION
 ENV CGO_ENABLED=0
 RUN make bin/compose-controller bin/compose-controller.test e2e-binary bin/installer bin/api-server bin/api-server.test bin/e2e_benchmark
-RUN go get github.com/onsi/ginkgo/ginkgo
+RUN GO111MODULE=on go install github.com/onsi/ginkgo/ginkgo@v1.16.5
 RUN curl -fLO https://storage.googleapis.com/kubernetes-release/release/v${KUBERNETES_VERSION}/bin/linux/amd64/kubectl && \
   chmod +x ./kubectl && \
   mv ./kubectl /bin/kubectl

--- a/Dockerfile.buildkit
+++ b/Dockerfile.buildkit
@@ -12,6 +12,7 @@ USER compose
 WORKDIR /home/compose
 
 FROM ${BUILD_BASE} AS build
+ENV GO111MODULE=off
 
 RUN apk add --no-cache \
   coreutils \

--- a/Dockerfile.buildkit
+++ b/Dockerfile.buildkit
@@ -32,7 +32,7 @@ ARG KUBERNETES_VERSION
 ENV GITCOMMIT=$GITCOMMIT VERSION=$VERSION BUILDTIME=$BUILDTIME IMAGE_REPO_PREFIX=$IMAGE_REPO_PREFIX KUBERNETES_VERSION=$KUBERNETES_VERSION
 ENV CGO_ENABLED=0
 RUN --mount=target=/root/.cache,type=cache make bin/compose-controller bin/compose-controller.test e2e-binary bin/installer bin/api-server bin/api-server.test bin/e2e_benchmark
-RUN go get github.com/onsi/ginkgo/ginkgo
+RUN GO111MODULE=on go install github.com/onsi/ginkgo/ginkgo@v1.16.5
 RUN curl -fLO https://storage.googleapis.com/kubernetes-release/release/v${KUBERNETES_VERSION}/bin/linux/amd64/kubectl && \
   chmod +x ./kubectl && \
   mv ./kubectl /bin/kubectl

--- a/Dockerfile.debuggable
+++ b/Dockerfile.debuggable
@@ -3,6 +3,7 @@ ARG BUILD_BASE
 ARG RUN_BASE
 
 FROM ${BUILD_BASE} AS build
+ENV GO111MODULE=off
 RUN apk add --no-cache \
   coreutils \
   make \

--- a/Makefile
+++ b/Makefile
@@ -75,7 +75,7 @@ check-licenses: ## Check the licenses for our dependencies
 	$(GOPATH)/bin/wwhrd check
 
 install-ginkgo:
-	go get -u github.com/onsi/ginkgo/ginkgo
+	GO111MODULE=on go install github.com/onsi/ginkgo/ginkgo@v1.16.5
 	go get -u github.com/onsi/gomega/...
 
 e2e: ## Run the e2e tests

--- a/docker.Makefile
+++ b/docker.Makefile
@@ -1,8 +1,8 @@
 include common.mk
 
-export BUILD_BASE_IMAGE = golang:1.13.0-alpine3.10
-export TEST_BASE_IMAGE = golang:1.13.0
-export RUN_BASE_IMAGE = alpine:3.10.2
+export BUILD_BASE_IMAGE = golang:1.16.13-alpine3.14
+export TEST_BASE_IMAGE = golang:1.16.13
+export RUN_BASE_IMAGE = alpine:3.14
 export KUBERNETES_VERSION ?= 1.16.1
 KIND_VERSION ?= 0.5.1
 IMAGES = ${IMAGE_REPO_PREFIX}controller ${IMAGE_REPO_PREFIX}controller-coverage ${IMAGE_REPO_PREFIX}e2e-tests ${IMAGE_REPO_PREFIX}e2e-benchmark ${IMAGE_REPO_PREFIX}api-server ${IMAGE_REPO_PREFIX}api-server-coverage ${IMAGE_REPO_PREFIX}installer

--- a/dockerfiles/Dockerfile.dev
+++ b/dockerfiles/Dockerfile.dev
@@ -1,6 +1,7 @@
 ARG     BUILD_BASE
 
 FROM    ${BUILD_BASE} as dev
+ENV     GO111MODULE=off
 
 RUN     apk add --no-cache \
         bash \

--- a/dockerfiles/Dockerfile.test
+++ b/dockerfiles/Dockerfile.test
@@ -1,5 +1,6 @@
 ARG     TEST_BASE
 FROM    ${TEST_BASE}
+ENV     GO111MODULE=off
 WORKDIR /go/src/github.com/docker/compose-on-kubernetes
 RUN apt update && apt install -y build-essential && rm -rf /var/lib/apt/lists/*
 RUN go get -u github.com/wadey/gocovmerge

